### PR TITLE
Remove Clang 11 dependencies as they conflict with Clang 10 dependencies

### LINKED ISSUE #2995: Broken Ubuntu 20 clang builds as of ~12 hours ago
**Description**  
My C++/Cmake project has been failing to build on github actions for the past 12 hours. see https://github.com/drewcassidy/quicktex/actions, build [121](https://github.com/drewcassidy/quicktex/actions/runs/664078687) succeeded before I reran it an hour ago to confirm it was the build environment causing my issues and not the code. It seems like `libc++-dev` and/or `libc++-11-dev` are not installed correctly.

**Area for Triage**:  
C/C++

**Question, Bug, or Feature?**:  
Bug

**Virtual environments affected**  
- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [x] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11.0
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Image version**
20210315.1

**Expected behavior**  
Clang build succeeds

**Actual behavior**  
Either a failure to find Libc++ (C++11 libraries are unable to be found) or a failure to find the abi (`cannot find -lc++abi`) when clang 11 is used

### PR DESCRIPTION
**Pull request by catthehacker**
Reverts actions/virtual-environments#2851 since it breaks already installed clang-10
Fixes #2995
Unblocks #2939

### CODE REVIEW

**Comment by miketimofeev** [comment]
@catthehacker it looks like we can leave clang 11 itself, just need to remove the additional libs and make a not in the readme that they need to be installed separately in order to use `-stdlib=libc++`

File: images/linux/scripts/SoftwareReport/SoftwareReport.Common.psm1 [hunk 1]
```diff
@@ -30,7 +30,8 @@ function Get-ClangVersions {
             $Matches.version
         }
     } | Sort-Object {[Version]$_}
-    return "Clang " + ($clangVersions -Join ", ")
+    $clangVersions = $clangVersions -Join ", "
+    return "Clang " + ($clangVersions -match '11.*' ? ($clangVersions + " (For Clang 11 you need to manually install required dependencies)") : $clangVersions)
```
Comment (miketimofeev) [inline images/linux/scripts/SoftwareReport/SoftwareReport.Common.psm1:34]: Actually, it looks like they need to be installed for all except the default one, like `libc++abi-9-dev` and `libc++-9-dev` for clang 9, so I'm not sure we need it in the readme. 🤔 @maxim-lobanov thoughts?

Comment (maxim-lobanov) [reply images/linux/scripts/SoftwareReport/SoftwareReport.Common.psm1:34]: May be we can don't write it in readme for now and see if we have more questions from customers. If it is unclear point- we can document it later

Comment (miketimofeev) [reply images/linux/scripts/SoftwareReport/SoftwareReport.Common.psm1:34]: @catthehacker remove it for now, please

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**